File: HorseRacing.Contracts/Models/Game/Responses/PlayGame/PlayGameResponse.cs
using HorseRacing.Contracts.Base.Responses;

namespace HorseRacing.Contracts.Models.Game.Responses.PlayGame
{
    public class PlayGameResponse : BaseResponse<PlayGameResponseDto>
    {
        public PlayGameResponse() : base() { }
        public PlayGameResponse(PlayGameResponseDto data) : base(data) { }
    }
}

File: HorseRacing.Web/src/interfaces/api/contracts/model/game/responses/play-game/play-game-response.ts
import type { BaseResponse } from '~/interfaces/api/contracts/base/responses/base-response';
import type { PlayGameResponseDto } from './play-game-response-dto';

export class PlayGameResponse implements BaseResponse<PlayGameResponseDto> {
    Data: PlayGameResponseDto;

    constructor(data?: PlayGameResponseDto) {
        this.Data = data || {
            GameId: '',
            Name: '',
            InitialDeck: [],
            HorseBets: [],
            Events: [],
        };
    }
}
